google-http-client-xml/src/test/java/com/google/api/client/xml/XmlEnumTest.java

```diff
@@ -1,64 +1,54 @@
 import org.xmlpull.v1.XmlSerializer;
 import com.google.api.client.util.Key;
 import com.google.api.client.util.Value;
-import junit.framework.TestCase;
-
-public class XmlEnumTest extends TestCase {
-
-  public enum AnyEnum {
-    @Value ENUM_1,
-    @Value ENUM_2
-  }
-
-  public static class AnyType {
-    @Key("@attr")
-    public Object attr;
-    @Key
-    public Object elem;
-    @Key
-    public Object rep;
-    @Key("@anyEnum")
-    public XmlEnumTest.AnyEnum anyEnum;
-    @Key
-    public XmlEnumTest.AnyEnum anotherEnum;
-    @Key
-    public ValueType value;
-  }
-
-  public static class AnyTypeEnumElementOnly {
-    @Key
-    public XmlEnumTest.AnyEnum elementEnum;
-  }
-
-  public static class AnyTypeEnumAttributeOnly {
-    @Key("@attributeEnum")
-    public XmlEnumTest.AnyEnum attributeEnum;
-  }
-
-  public static class ValueType {
-    @Key("text()")
-    public XmlEnumTest.AnyEnum content;
-  }
-
-  private static final String XML =
-      "<?xml version=\"1.0\"?><any anyEnum=\"ENUM_1\" attr=\"value\" xmlns=\"http://www.w3.org/2005/Atom\">"
-          + "<anotherEnum>ENUM_2</anotherEnum><elem>content</elem><rep>rep1</rep><rep>rep2</rep><value>ENUM_1</value></any>";
-
-  private static final String XML_ENUM_ELEMENT_ONLY =   "<?xml version=\"1.0\"?><any xmlns=\"http://www.w3.org/2005/Atom\"><elementEnum>ENUM_2</elementEnum></any>";
-
-  private static final String XML_ENUM_ATTRIBUTE_ONLY =   "<?xml version=\"1.0\"?><any attributeEnum=\"ENUM_1\" xmlns=\"http://www.w3.org/2005/Atom\" />";
-
-  private static final String XML_ENUM_INCORRECT =   "<?xml version=\"1.0\"?><any xmlns=\"http://www.w3.org/2005/Atom\"><elementEnum>ENUM_3</elementEnum></any>";
 
+/**
+ * Tests {@link Xml}.
+ *
+ * @author Gerald Madlmayr
+ */
+public class XmlEnumTest {
+
+  private static final String XML = "<?xml version=\"1.0\"?><any anyEnum=\"ENUM_1\" attr" +
+      "=\"value\" xmlns=\"http://www.w3.org/2005/Atom\"><anotherEnum>ENUM_2</anotherEnum" +
+      "><elem>content</elem><rep>rep1</rep><rep>rep2</rep><value>ENUM_1</value></any>";
+  private static final String XML_ENUM_ELEMENT_ONLY = "<?xml version=\"1.0\"?><any xmlns" +
+      "=\"http://www.w3.org/2005/Atom\"><elementEnum>ENUM_2</elementEnum></any>";
+  private static final String XML_ENUM_ATTRIBUTE_ONLY = "<?xml version=\"1.0\"?><any " +
+      "attributeEnum=\"ENUM_1\" xmlns=\"http://www.w3.org/2005/Atom\" />";
+  private static final String XML_ENUM_INCORRECT = "<?xml version=\"1.0\"?><any xmlns=\"http" +
+      "://www.w3.org/2005/Atom\"><elementEnum>ENUM_3</elementEnum></any>";
+  private static final String XML_ENUM_ELEMENT_ONLY_NESTED = "<?xml version=\"1.0\"?><any " +
+      "xmlns=\"http://www.w3.org/2005/Atom\"><elementEnum>ENUM_2<nested>something</nested" +
+      "></elementEnum></any>";
 
   @SuppressWarnings("cast")
-  public void testParse_anyType() throws Exception {
+  @Test
+  public void testParseAnyType() throws Exception {
     AnyType xml = new AnyType();
```

Comment:
this variable looks unused

Reply:
No exactly sure, which variable you refer to. The ```xml``` is used during validation. Could you pls clarified this?

Reply:
fixed, by removing irrelvant annotation.